repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphRetriever.java

```diff
@@ -740,7 +741,9 @@ private Object mapVertexToAttribute(AtlasVertex entityVertex, AtlasAttribute att
         Map<String, AtlasObjectId> ret        = null;
         Map                        softRefVal = entityVertex.getProperty(attribute.getVertexPropertyName(), Map.class);
 
-        if (MapUtils.isNotEmpty(softRefVal)) {
+        if (softRefVal == null || MapUtils.isEmpty(softRefVal)) {
```

Comment:
MapUtils.isEmpty(softRefVal) checks for null and empty map, consider recoming softRefVal == null from if condition